### PR TITLE
Commit for 1877-sslHostNameVerify

### PR DESCRIPTION
**Pull request by dburdine**

### CODE REVIEW

**Comment by dburdine** [comment]
fixes #1877

**Comment by dburdine** [comment]
I tested these changes manually in a WLP runtime. I verified System property com.ibm.ssl.performURLHostNameVerification=true and that on the handshake the proper extension was added: Extension server_name, server_name: [host_name: www.ibm.com]

File: dev/com.ibm.ws.ssl/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -20,6 +20,8 @@ default.sslRef=Default SSL repertoire
 default.sslRef.desc=The name of the SSL repertoire that will be used as the default.  The default SSL repertoire called defaultSSLConfig is used if none is specified.
 default.outboundSSLRef=Outbound default SSL repertoire
 default.outboundSSLRef.desc=The name of the SSL repertoire that will be used as the default for outbound SSL connections.  The default SSL repertoire will be used if no SSL repertoire is specified.  This attribute is ignored if the transportSecurity-1.0 feature is not specified.
+default.httpHostNameVerification=Perform hostname verification for http outbound requests.
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.ssl/resources/OSGI-INF/l10n/metatype.properties]: Run host name verification for HTTP outbound requests.

File: dev/com.ibm.ws.ssl/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -20,6 +20,8 @@ default.sslRef=Default SSL repertoire
...
+default.httpHostNameVerification.desc=Specifies whether SSL outbound hostname verification is enabled. The default is false. If set to true then during the outbound SSL handshake requests the JDK will perform verification that the certificate returned contains a hostname that matches the outbound http request.
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.ssl/resources/OSGI-INF/l10n/metatype.properties]: Specifies whether SSL outbound host name verification is enabled. The default is false. If set to true, then during the outbound SSL handshake requests, the JDK will verify that the returned certificate contains a host name that matches the outbound HTTP request.

**Comment by dburdine** [comment]
#build

**Comment by wrodrig** [comment]
#build

**Comment by dburdine** [comment]
#build

**Comment by dburdine** [comment]
#build

**Comment by dburdine** [comment]
#build

**Comment by wrodrig** [comment]
#build

**Comment by dburdine** [comment]
#build

**Comment by dburdine** [comment]
#build

**Comment by dburdine** [comment]
#build

**Comment by dburdine** [comment]
#build

**Comment by dburdine** [comment]
The builds are failing with defects that are not related to my code change. Here are the errors found in build:
https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_n0iVcA9zEeiStsw2He7iFw

100% testSecureAsyncObserverUsingRunAsWithAuthority
timeout
Test Failure (20180124-1945): com.ibm.ws.cdi20.fat.tests.SecureAsyncEventsTest.testSecureAsyncObserverUsingRunAsWithAuthority (250640)

97%  testSecureAsyncObserverUsingRunAsWithNOAuthority
timeout
Test Failure (20180124-1945): com.ibm.ws.cdi20.fat.tests.SecureAsyncEventsTest.testSecureAsyncObserverUsingRunAsWithAuthority (250640)


93% com.ibm.ws.cdi12.fat.tests.AppExtensionTest
Test Failure (20180212-2037): junit.framework.TestSuite.com.ibm.ws.cdi12.fat.tests.AppExtensionTest (251322)

100% ApplicationUpdate_StopServer
Test Failure (20170702-0123): com.ibm.ws.collective.routing.controller.fat.RoutingInfoManagerTest.ApplicationUpdate_StopServer (243880)


97% com.ibm.ws.microprofile.config.fat.tests.DynamicSourcesTest
Test Failure (20180202-1159): junit.framework.TestSuite.com.ibm.ws.microprofile.config.fat.tests.DynamicSourcesTest (250966)

**Pull request merged**